presto-main/src/main/java/com/facebook/presto/operator/aggregation/MaxAggregation.java

```diff
@@ -0,0 +1,397 @@
+            }
+
+            @Override
+            public void setLong(long value)
+            {
+                values.set(getGroupId(), value);
+            }
+
+            @Override
+            public boolean isNull()
+            {
+                return nulls.get(getGroupId());
+            }
+
+            @Override
+            public void setNull(boolean value)
+            {
+                nulls.set(getGroupId(), value);
+            }
+        }
+
+        public static class SingleLongState
+                implements LongState
+        {
+            private final Type type;
+            private final MethodHandle methodHandle;
+            private long value;
+            private boolean nul;
+
+            public SingleLongState(Type type, MethodHandle methodHandle)
+            {
+                this.type = type;
+                this.methodHandle = methodHandle;
+                this.value = 0;
+                this.nul = true;
+            }
+
+            @Override
+            public long getEstimatedSize()
+            {
+                return 16;
+            }
+
+            @Override
+            public Type getType()
+            {
+                return type;
+            }
+
+            @Override
+            public MethodHandle getMethodHandle()
+            {
+                return methodHandle;
+            }
+
+            @Override
+            public long getLong()
+            {
+                return value;
+            }
+
+            @Override
+            public void setLong(long value)
+            {
+                this.value = value;
+            }
+
+            @Override
+            public boolean isNull()
+            {
+                return nul;
+            }
+
+            @Override
+            public void setNull(boolean value)
+            {
+                this.nul = value;
+            }
+        }
+    }
+
+    private static class StateSerializerForLong
+            implements AccumulatorStateSerializer<LongState>
+    {
+        @Override
+        public Type getSerializedType()
+        {
+            return VARCHAR;
+        }
+
+        @Override
+        public void serialize(LongState state, BlockBuilder out)
+        {
+            SliceOutput sliceOutput = new DynamicSliceOutput((int) state.getEstimatedSize());
+            sliceOutput.writeLong(state.getLong());
+            sliceOutput.writeBoolean(state.isNull());
+            Slice slice = sliceOutput.slice();
+            out.writeBytes(slice, 0, slice.length());
+            out.closeEntry();
+        }
+
+        @Override
+        public void deserialize(Block block, int index, LongState state)
+        {
+            SliceInput input = block.getSlice(index, 0, block.getLength(index)).getInput();
+            long value = input.readLong();
+            state.setLong(value);
+            boolean nul = input.readBoolean();
+            state.setNull(nul);
+        }
+    }
+
+    public static class ForLong
```

Comment:
Move this to a top level class like MinMaxAggregationFunctions, so that we can reuse it for min()